src/ORM/Connect/Database.php

```diff
@@ -343,12 +343,8 @@ public function manipulate($manipulation)
                         $query->addWhere(array('"ID"' => $writeInfo['id']));
                     }
 
-                    // Test to see if this update query shouldn't, in fact, be an insert
-                    if ($query->toSelect()->count()) {
```

Comment:
`affectedRows()` seems to return false if you run an update query that doesn't change any columns.

Well, I know that it erroneously returns false sometimes, I'm guessing that that's the reason why.